client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/iteration/RemoteInnerPublisherHandler.java

```diff
@@ -94,7 +94,7 @@ protected long handleNextResponse(IterationNextResponse<K, E> nextResponse, Map.
          targetComplete();
       }
       IntSet completedSegments = nextResponse.getCompletedSegments();
-      if (completedSegments != null && log.isTraceEnabled()) {
+      if (completedSegments != null) {
```

Comment:
The `isTraceEnabled` seems misplaced here.

Reply:
This change is not needed. The targetSegments are only fully used if tracing is enabled. Otherwise it is treated as an immutable collection.